src/vm/mod.rs

```diff
@@ -335,7 +335,7 @@ fn eval_all(
                     global_context.execute(|global_context| {
                         let mut call_stack = CallStack::new();
                         let mut env = Environment::new(
-                            global_context, contract_context, &mut call_stack, Some(publisher.clone()), Some(publisher.clone()));
+                            global_context, contract_context, &mut call_stack, Some(publisher.clone()), Some(publisher.clone()), None);
```

Comment:
`eval_all` is used for the initialization of a new contract (which should probably be pointed out somewhere in `eval_all`'s rustdocs) -- currently, the "tx-sender" for a contract initialization is the contract "publisher", which is always the origin account that broadcasted the contract publish tx. If you want `tx-sponsor?` to work during contract initialization as well, rather than passing a `None` here, I think we'd need to pass through the sponsor from the transaction.